### PR TITLE
feat(fun): random animal CC

### PR DESCRIPTION
**Pull request by GenryMg**
Is this how you PR?

### CODE REVIEW

**Comment by GenryMg** [comment]
Ok, I added a Meme making command, and some animal API comamnds. Feel free to DM/Ping/Respond with issues or things I need to fix. I will fix them ASAP.

**Comment by GenryMg** [comment]
Also, uh, don't ask about PR Merge from my repo to my repo... 
I'm new to github lol.

File: fun/Animal.cc.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{{ sendMessage nil (cembed "title" "Have a http cat!" "image" (sdict "url" $link)) }}
+{{else}}
+{{sendMessage nil "Please provide an http response"}}
+{{end}}
+{{else if eq $animal "cat"}}
+{{$link := joinStr "" "https://cataas.com/cat?filter=" (urlescape adjective)}}
+{{$ID := sendMessageRetID nil (cembed "image" (sdict "url" $link))}}
+{{sleep 2}}
+{{$msg := getMessage nil $ID}}
+{{editMessage nil $ID (cembed "title" "Have a cat!" "image" (sdict "url" (index $msg.Embeds 0).Image.ProxyURL))}}
```
Comment (Jo3-L) [inline fun/Animal.cc.lua]: Why is there a need for this edit? Seems to work fine for me without.

**Comment by Jo3-L** [comment]
Also, would prefer you split these into two separate CCs next time (meme and animal).

**Comment by GenryMg** [comment]
The two CCs should be separate unless I did something wrong with making the new files.

File: fun/Animal.cc.lua (hunk 1 above)
Comment (GenryMg) [reply fun/Animal.cc.lua]: uhhh. Satty wrote that one some time ago, I don't know actually.

Comment (GenryMg) [reply fun/Animal.cc.lua]: I asked him, and he said it's becuase of discord not caching correctly, causing formatting problems and problems with the output. I tested it without the sleep, and sometimes it sneds one cat but then displays another when you click on it.

File: fun/Animal.cc.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{{/* This is an API command, where you can generate pictures of ducks, foxes, goats, cats, shibas, and my personal favourite, http Cats. I will probably add more afterwards, but this is it for now
+     Recommended Trigger type: Command trigger, Trigger: animal.
+     Please note that the animal names are case sensitive, the first letter of the animal is capital. If you have any animal pictures or APIs, feel free to DM me (GenryMg#0001) I'll add them here.
```
[the remark this answers is not in the document]
Comment (GenryMg) [reply fun/Animal.cc.lua]: yeah, I realised I didn't format this sentence correctly. I meant that `-animal Fox` wouldn't work, but `-animal fox` would. I'm gonna change this

File: fun/Animal.cc.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{{$link := index $links (randInt (len $links)) }}
+{{sendMessage nil (cembed "title" "Have a shiba!" "image" (sdict "url" $link))}}
+{{else if eq $animal "help"}}
+{{$col := randInt 111111 999999}}
```
[the remark this answers is not in the document]
Comment (GenryMg) [reply fun/Animal.cc.lua]: ok, I didn't realise this.

File: fun/Animal.cc.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
...40b4ca2512a8d2df6a7fce0.jpg" "http://cdn.shibe.online/shibes/1dc7b520e8c08cfaf7efaeb7cc3b79ea8c6dc520.jpg" "http://cdn.shibe.online/shibes/3c2ce582117afea3680ed279943b4b7081bcbdef.jpg" "http://cdn.shibe.online/shibes/91d21aaa33a55f414c2a3488b7c0c155dc045b1f.jpg" "http://cdn.shibe.online/shibes/a870fc6e9e51b92cefb3f7f945190bc0b29060b3.jpg" "http://cdn.shibe.online/shibes/d5950fd9ab48f73e5693b055146f77bc5d2f648b.jpg" "http://cdn.shibe.online/shibes/eb2bbc42d836ad1b6589a4e407452415400f6222.jpg" "http://cdn.shibe.online/shibes/2e2a074e2f1756decf9e84da095d60e19cb1d982.jpg" "http://cdn.shibe.online/shibes/80048945e625d11210044e516c4a0e74b2b361da.jpg" "http://cdn.shibe.online/shibes/9794958daf9c3f8a6997ece8ca3434d0c07f3e7b.jpg" "http://cdn.shibe.online/shibes/be2cb2c635ddf024497f5e7a8c73398974782f0c.jpg" "http://cdn.shibe.online/shibes/5fcdfb9dae06fff466a45005523a46f6ad022b4e.jpg" "http://cdn.shibe.online/shibes/076bc996fec3ddc0d7aee52d2150ad8749a95b31.jpg" "http://cdn.shibe.online/shibes/b1f4767ccbe873a357ab9179a44d6e1ae11c1b6f.jpg" "http://cdn.shibe.online/shibes/9eab8417d39237e7c6d726aba9d8f688ed55e307.jpg" "http://cdn.shibe.online/shibes/45861677217a38e7495dd8a8a3f97b76860171b3.jpg"}}
```
[the remark this answers is not in the document]
Comment (GenryMg) [reply fun/Animal.cc.lua]: good point. I will work on that

**Comment by GenryMg** [comment]
I will close this PR.

**Pull request closed without merging**